### ISSUE TITLE
MonoDevelop closes trying to open a solution project

### ISSUE THREAD

**Issue by netonjm** [mono/monodevelop#4032]
I tried to rename a project referencing a SharedProject. 

MonoDevelop finishing this action did an exception and now  I can't open this project anymore

This happens in alpha channel.

Full log:
https://gist.github.com/netonjm/1bd754a1ff31f92285591e1227d288b7

Visual Studio Enterprise 2017 for Mac (Preview)
Version 7.4 Preview (7.4 build 1026)
Installation UUID: ac2f5f5a-65c6-4e02-aae1-61d2ac9aabc1
Runtime:
	Mono 5.8.0.127 (2017-10/1eac626536b) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)

	Package version: 508000127

NuGet
Versión: 4.3.1.4445

.NET Core
Runtime: /usr/local/share/dotnet/dotnet
Versiones de tiempo de ejecución:
	2.0.5
	2.0.3
	2.0.0
	1.1.2
	1.1.1
	1.0.5
	1.0.4
	1.0.3
SDK: /usr/local/share/dotnet/sdk/2.1.4/Sdks
Versiones del SDK:
	2.1.4
	2.0.3
	2.0.0
	1.0.3
	1.0.1
	1.0.0-preview2-003156
SDK de MSBuild: /Library/Frameworks/Mono.framework/Versions/5.8.0/lib/mono/msbuild/15.0/bin/Sdks

Xamarin.Profiler
Versión: 1.6.3
Ubicación: /Users/jmedrano/Projects/profiler/src/XamarinProfiler.Mac/bin/Debug/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Xamarin.Android
Versión: 8.2.0.15 (Visual Studio Enterprise)
Android SDK: /Users/jmedrano/Library/Android/sdk
	Versiones de Android admitidas:
		4.4 (nivel de API 19)
		7.1 (nivel de API 25)
		8.0 (nivel de API 26)

Versión de las herramientas del SDK: 26.1.1
Versión de las herramientas de plataforma del SDK: 27.0.1
Versión de las herramientas de compilación del SDK: 26.0.3

Java SDK: /usr
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

El código EPL de Android Designer está disponible EPL:
https://github.com/xamarin/AndroidDesigner.EPL

Apple Developer Tools
Xcode 9.2 (13772)
Build 9C40b

Xamarin.Mac
Version: 4.2.0.20 (Visual Studio Enterprise)

Xamarin.iOS
Version: 11.8.0.20 (Visual Studio Enterprise)
Hash: 1c6f300e
Branch: d15-6
Build date: 2018-02-07 23:42:40-0500

Xamarin Inspector
Version: 1.4.0
Hash: b3f92f9
Branch: master
Build date: Fri, 19 Jan 2018 22:00:34 GMT
Client compatibility: 1

Build Information
Release ID: 704001026
Git revision: 56e88d4d0e790e77e1414b8fed22832260fb244f
Build date: 2018-02-16 13:51:48-05
Xamarin addins: 593b39138609e1e4fa0da4c1a1cdfecc97a285a5
Build lane: monodevelop-lion-d15-6

Operating System
Mac OS X 10.13.3
Darwin 17.4.0 Darwin Kernel Version 17.4.0
    Sun Dec 17 09:19:54 PST 2017
    root:xnu-4570.41.2~1/RELEASE_X86_64 x86_64

Enabled user installed addins
Gist Ide Information 1.1.0
AddinMaker 1.4.2
NuGet Package Explorer 0.2
NuGet Package Management Extensions 0.12.2
Redth's Addins 1.0.9
Show in Github 1.0.4
Internet of Things (IoT) development (Preview) 9.2

> VS bug #576053

**Comment by netonjm** [comment]
[avalonia.zip](https://github.com/mono/monodevelop/files/1778345/avalonia.zip)

Try to open the solution

**Comment by mrward** [comment]
Null reference in the CSharpLineSeparatorService in the causing a fatal unhandled exception.

```
FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object
  at Microsoft.CodeAnalysis.Editor.CSharp.LineSeparator.CSharpLineSeparatorService+<GetLineSeparatorsAsync>d__0.MoveNext () [0x00011] in /Users/builder/data/lanes/5810/56e88d4d/source/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Features/LineSeparators/CSharpLineSeparatorService.cs:30 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:152 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:357 
  at MonoDevelop.Ide.Editor.Extension.LineSeparatorTextEditorExtension+<DocumentContext_DocumentParsed>d__8.MoveNext () [0x000ad] in /Users/builder/data/lanes/5810/56e88d4d/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/LineSeparatorTextEditorExtension.cs:95 
```

**Comment by Therzok** [comment]
Seems like AnalysisDocument is null.

**Comment by mrward** [comment]
Whilst we could check for AnalysisDocument is null - should we also just have a try/catch around the DocumentContext_DocumentParsed code so unhandled exceptions are caught and logged instead of causing the IDE to crash/terminate?

**Comment by mrward** [comment]
In latest d15-7 and master this code would run on the UI thread. Not sure if would have different behaviour for an unhandled exception.

The about dialog indicates this is in d15-6 so do we want to fix it there too?

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-33] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #4037 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 2986e4bcefb663d364b99dc4cea686b3a0903712 (xamarin/monodevelop)
